File: src/cleans/clean_chars.py
import sys

print sys.argv[1]

with open(sys.argv[1]) as f:
    raw = f.readlines()

REPS = [{'utf8': "\xE2\x80\x9C", 'rep': '"'},
        {'utf8': "\xE2\x80\x9D", 'rep': '"'},
        {'utf8': "\xE2\x80\x93", 'rep': '--'},
        {'utf8': "\xE2\x80\xA6", 'rep': '...'},
        {'utf8': "\xE2\x80\x98", 'rep': "'"},

        {'utf8': "\xE2\x80\x99", 'rep': "'"}

        ]


def checkString(s):
    for x in xrange(len(s)):
        if ord(s[x]) > 127:
            for y in xrange(5):
                print ord(s[x + y])
            return False
    return True

for x in xrange(len(raw)):
    # if '\xE2\x80\x73' in raw[x]:
    #    print raw[x]
    r = raw[x]
    for o in REPS:
        r = r.replace(o["utf8"], o["rep"])
    raw[x] = r
    if not checkString(r):
        print "OOPS"
        print "::" + r + "::"
        break

with open("New.mark", "w") as f:
    f.writelines(raw)

File: src/cleans/CleanChars.java
package cleans;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.InputStream;
import java.io.OutputStream;
import java.util.ArrayList;
import java.util.List;

public class CleanChars {
	
	public static void main(String [] args) throws Exception {
		
		String fname = "content/Arcade/SpaceInvaders/SpaceInvaders.mark";
		
		InputStream is = new FileInputStream(fname);
		int [] data = new int[is.available()];
		for(int x=0;x<data.length;++x) {
			data[x] = is.read();
		}
		is.close();
		
		List<Integer> newData = new ArrayList<Integer>();
		for(int x=0;x<data.length;++x) {
			//System.out.println(data[x]);
			if(data[x]>127) {
				if(data[x]==0xE2 && data[x+1]==0x80) {
					switch(data[x+2]) {
					case 0x9C:
						newData.add((int) '"');
						break;
					case 0x9D:
						newData.add((int) '"');
						break;
					case 0x93:
						newData.add((int) '"');
						break;
					case 0xA6:
						newData.add((int) '"');
						break;
					case 0x98:
						newData.add((int) '"');
						break;
					case 0x99:
						newData.add((int) '"');
						break;
					default:
						throw new Exception("Unknown "+data[x+2]);
					}
					x = x + 2;
				} else {
					throw new Exception("Unknown "+data[x]);
				}				
			} else {
				newData.add(data[x]);
			}
		}		
		
		OutputStream os = new FileOutputStream("New.mark");
		for(int i : newData) {
			os.write(i);
		}
		os.flush();
		os.close();
		
	}

}
